### PR TITLE
Some Kafka client improvements

### PR DESCRIPTION
**Pull request by yangxi**
* Add TLS, compression type, and exactly once support
* Set the partition and the timestamp in Kafka record to none
* Remove the unused partition and partition distribution fields
* Set the key to None when the key_len is 0

### CODE REVIEW

File: configs/kafka.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ protocol = "kafka"
 # the interval for stats integration and reporting
 interval = 1
 # the number of intervals to run the test for
-duration = 300
+duration = 30
```
Comment (brayniac) [inline configs/kafka.toml:9]: I don't think we should reduce this in the example

File: src/workload/mod.rs [hunk 2]
```diff
@@ -411,19 +409,19 @@ pub struct Topics {
...
     subscriber_concurrency: usize,
+    kafka_single_subscriber_group: bool,
 }
 
 impl Topics {
     pub fn new(config: &Config, topics: &config::Topics) -> Self {
         let message_random_bytes =
             estimate_random_bytes_needed(topics.message_len(), topics.compression_ratio());
-
+        println!("random bytes: {}", message_random_bytes);
```
Comment (brayniac) [inline src/workload/mod.rs:424]: This should be removed

File: src/config/pubsub.rs [hunk 3]
```diff
@@ -28,7 +38,14 @@ pub struct Pubsub {
     write_buffer_size: usize,
 
     // kafka specific configs
-    kafka_acks: Option<String>,
+    #[serde(default)]
+    kafka_exactly_once: bool,
+    #[serde(default = "default_kafka_acks")]
+    kafka_acks: String,
+    #[serde(default = "default_kafka_compression_type")]
+    kafka_compression_type: String,
```
Comment (brayniac) [inline src/config/pubsub.rs:46]: should this be an `Option<String>` instead?

**Comment by yangxi** [comment]
* Add TLS, compression type, and exactly-once support
* Set the partition and the timestamp in Kafka record to none
* Remove the unused partition and partition distribution fields
* Set the key to None when the key_len is 0

**Pull request merged**